plugins/transport/socket/main.go

```diff
@@ -81,6 +81,7 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 	go func() {
 		for {
 			n, err := pc.Read(msgBuffer)
+			msgBuffer[n] = 0
```

Comment:
You found a pretty critical bug here that exists in master right now (woops). Could you replace the call to the write function with `w(mspBuffer[:n])` instead of setting a null value? Just so the write function stack size can be minimized when the buffer isn't full

Reply:
Props on catching that, I don't know how that made it past all the testing I did, but I guess that is why unit tests are so important (plus a working CI pipeline)

Reply:
Kudos indeed. Just out of curiosity (since I have no idea what's wrong here), what the last byte contains?

Reply:
The message buffer isn't cleared after a message is consumed, so if a new, shorter message arrives, there can be remnants of the old one left after the new message ends. Since the socket plugin was writing the whole buffer to the handler every time, the handler would get corrupted data

Reply:
Also assume this is `w(msgBuffer[:n])` without the typo heh

Reply:
If only comments didn't tell everyone that you edited it